libraries/AP_PiccoloCAN/AP_PiccoloCAN.cpp

```diff
@@ -136,13 +175,22 @@ void AP_PiccoloCAN::loop()
         // 1ms loop delay
         hal.scheduler->delay_microseconds(1 * 1000);
 
-        // Transmit CAN commands at regular intervals
-        if (txCounter++ > CMD_TX_PERIOD) {
+        // Transmit ESC commands at regular intervals
+        if (esc_tx_counter++ > _esc_ms) {
+            esc_tx_counter = 0;
 
-            txCounter = 0;
+            if (_esc_ms >= PICCOLO_MSG_RATE_MS_MIN) {
+                send_esc_messages();
+            }
+        }
+
+        // Transmit servo commands at regular intervals
+        if (srv_tx_counter++ > _srv_ms) {
+            srv_tx_counter = 0;
 
-            // Transmit ESC commands
-            send_esc_messages();
+            if (_srv_ms >= PICCOLO_MSG_RATE_MS_MIN) {
+                // TODO - Transmit servo messages
```

Comment:
shouldn't add the params till we can use them

Reply:
This has now been removed from this PR